### ISSUE TITLE
Support batch/minibatch gradient descent

### ISSUE THREAD

**Issue by martelloni** [MusicallyEmbodiedML/MLP_XMOS#2]
`Loss.h` line 24: Only SGD is supported - support batch/minibatch GD
Problem is that minibatch requires total rewrite of backprop. I don't know how to handle activation functions for each example and how they relate to backprop.
